crates/context/config/src/client/protocol/evm.rs

```diff
@@ -159,35 +162,27 @@ impl Network {
         method: String,
         args: Vec<u8>,
     ) -> Result<Vec<u8>, EvmError> {
-        // Create RPC client
-        let client: ReqwestClient = ClientBuilder::default().http(self.client.parse().unwrap());
-
-        // Parse contract address
         let address = contract_id
             .parse::<Address>()
             .map_err(|e| EvmError::Custom {
                 operation: ErrorOperation::Mutate,
                 reason: e.to_string(),
             })?;
 
-        // Create method selector (first 4 bytes of method signature hash)
         let method_selector = &keccak256(method.as_bytes())[..4];
 
-        // Combine method selector with arguments
         let call_data = [method_selector, &args].concat();
 
-        // Create parameters for the RPC call
         let params = vec![
             json!({
                 "to": address,
                 "data": Bytes::from(call_data)
             }),
             json!("latest"),
-        ]; // Added "latest" block parameter
+        ];
 
-        // Send the eth_call and get the response
         let response: Bytes =
-            client
+            self.client
                 .request("eth_call", params)
```

Comment:
were these considered:

1. [`Provider::call`](https://docs.rs/alloy/latest/alloy/providers/trait.Provider.html#method.call)
2. [`TransactionRequest::to`](https://docs.rs/alloy/latest/alloy/rpc/types/struct.TransactionRequest.html#method.to) & [`TransactionRequest::input`](https://docs.rs/alloy/latest/alloy/rpc/types/struct.TransactionRequest.html#method.input)
3. [`TransactionInput: From<Bytes>`](https://docs.rs/alloy/latest/alloy/rpc/types/struct.TransactionInput.html#impl-From%3CBytes%3E-for-TransactionInput)

should come down to something like this:

```rust
let request = TransactionRequest::default()
    .to(address)
    .input(Bytes::from(bytes).into());

let bytes = self.provider.call(request).await.map_err(...)?;
```